src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceServiceTest.kt

```diff
@@ -139,6 +139,22 @@ class LicenceServiceTest {
     assertThat(licence.nomsId).isEqualTo(aLicenceEntity.nomsId)
     assertThat(licence.bookingId).isEqualTo(aLicenceEntity.bookingId)
     assertThat(licence.pnc).isEqualTo(aLicenceEntity.pnc)
+    assertThat(licence.isInPssPeriod).isFalse()
+
+    verify(licenceRepository, times(1)).findById(1L)
+  }
+
+  @Test
+  fun `service transforms key fields of a licence in PSS period object correctly`() {
```

Comment:
You're LicenceEntityTest would then not need to worry about calling a service or calling a repository - you can get straight to the nitty gritty of testing the logic itself. 

It would also be good to test each branch in the conditions in the code 
```
  if (led != null && tused != null) {
    return led.isBefore(LocalDate.now()) && !(tused.isBefore(LocalDate.now()))
  }
  return false
```

what if led is null?
what if tused is null?
what if both are null?
what if led.isBefore now only 
what if tused.isBefore now only 
what if both are before now?

Reply:
added all cases for unit testing isInPSSPeriod